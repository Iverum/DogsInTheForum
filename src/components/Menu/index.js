import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router'
import UserMenu from '../User/menu'

export class Menu extends Component {

  render() {
    return (
      <header>
        <Link to='/'><h1>Dogs in the Forum</h1></Link>
        <UserMenu user={this.props.user} />
      </header>
    )
  }

}

export default connect(state => ({user: state.user}))(Menu)
