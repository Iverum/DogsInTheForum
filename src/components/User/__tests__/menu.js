import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Menu from '../menu'

test('user menu is rendered correctly', () => {
  const component = renderer.create(
    <Menu user={{ name: 'Test User' }}/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
