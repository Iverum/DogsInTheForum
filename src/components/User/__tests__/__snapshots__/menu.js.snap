exports[`test user menu is rendered correctly 1`] = `
<div
  className="User">
  <a
    className="button"
    onClick={[Function]}
    style={Object {}}>
    Test User
  </a>
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Log out" />
</div>
`;
